src/test/java/com/opipo/codechallenge/api/TransactionApiDelegateImplTest.java

```diff
@@ -0,0 +1,119 @@
+    void givenTransactionThenSaveIt() {
+        Transaction transaction = Mockito.mock(Transaction.class);
+        String iban = "myIban";
+        Mockito.when(transaction.getAccountIban()).thenReturn(iban);
+        HttpStatus expectedStatus = HttpStatus.CREATED;
+
+        Mockito.when(transactionService.validateIban(iban)).thenReturn(true);
+
+        ResponseEntity<Void> response = transactionApiDelegate.addTransaction(transaction);
+        Mockito.verify(transactionService).save(transaction);
+
+        assertEquals(expectedStatus, response.getStatusCode());
+    }
+
+    @Test
+    @DisplayName("Given transaction that provokes exception then return exception")
+    void givenTransactionThenThrowsException() {
+        Transaction transaction = Mockito.mock(Transaction.class);
+        String iban = "myIban";
+        Mockito.when(transaction.getAccountIban()).thenReturn(iban);
+
+        Mockito.when(transactionService.validateIban(iban)).thenReturn(true);
+        Mockito.doThrow(new IllegalArgumentException()).when(transactionService).save(transaction);
+
+        assertThrows(IllegalArgumentException.class, () -> transactionApiDelegate.addTransaction(transaction));
+
+    }
+
+    @Test
+    @DisplayName("Given accountIban and order then get list ordered")
+    void givenAccountIbanAndOrderThenReturnList() {
+        String iban = "myIban";
+        SortType sort = SortType.ASC;
+        List<Transaction> transactions = new ArrayList<>();
+        Mockito.when(transactionService.validateIban(iban)).thenReturn(true);
+        Mockito.when(transactionService.list(iban, sort)).thenReturn(transactions);
+
+        ResponseEntity<List<Transaction>> actual = transactionApiDelegate.getTransactions(iban, sort.toString());
+
+        assertNotNull(actual);
+        assertEquals(HttpStatus.OK, actual.getStatusCode());
+        assertEquals(transactions, actual.getBody());
+    }
+
+    @Test
+    @DisplayName("Given accountIban and no order then get list unordered")
+    void givenAccountIbanAndNoOrderThenReturnList() {
+        String iban = "myIban";
+        List<Transaction> transactions = new ArrayList<>();
+        Mockito.when(transactionService.validateIban(iban)).thenReturn(true);
+        Mockito.when(transactionService.list(iban, null)).thenReturn(transactions);
+
+        ResponseEntity<List<Transaction>> actual = transactionApiDelegate.getTransactions(iban, null);
+
+        assertNotNull(actual);
+        assertEquals(HttpStatus.OK, actual.getStatusCode());
+        assertEquals(transactions, actual.getBody());
+    }
+
+    @Test
+    @DisplayName("Given accountIban and bad order then throw exception")
+    void givenAccountIbanAndBadOrderThenThrowException() {
+        String iban = "myIban";
+        Mockito.when(transactionService.validateIban(iban)).thenReturn(true);
+        assertThrows(InvalidRequestException.class, () -> transactionApiDelegate.getTransactions(iban, "fake"));
+    }
+
+    @Test
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/albertinizao/codechallenge/pullRequest?prid=4681930)